src/object/srv_obj_migrate.c

```diff
@@ -1382,19 +1452,21 @@ migrate_enum_unpack_cb(struct dss_enum_unpack_io *io, void *data)
 		/* Convert EC object offset to DAOS offset. */
 		for (i = 0; i <= io->ui_iods_top; i++) {
 			daos_iod_t *iod = &io->ui_iods[i];
-			daos_recx_t *recxs;
-			int nr = 0;
 
 			rc = obj_recx_ec2_daos(oca, io->ui_oid.id_shard,
-					       iod->iod_recxs, iod->iod_nr,
-					       &recxs, &nr);
+					       &iod->iod_recxs, &iod->iod_nr);
 			if (rc != 0)
 				return rc;
 
-			if (iod->iod_recxs != recxs) {
-				D_FREE(iod->iod_recxs);
-				iod->iod_recxs = recxs;
-				iod->iod_nr = nr;
+			/* For data shard, convert to single shard offset */
+			if (arg->arg->shard < obj_ec_data_tgt_nr(oca)) {
```

Comment:
ec_2daos will convert the offset to the whole stripe offset, but it only rebuilds single shard here,  so it will pull out the single shard extent from the whole stripe.

Reply:
ok, i see, thanks.